pennylane/tape/qscript.py

```diff
@@ -1293,10 +1293,10 @@ def to_openqasm(self, wires=None, rotations=True, measure_all=True, precision=No
 
         return qasm_str
 
-    @staticmethod
-    def from_queue(queue):
+    @classmethod
+    def from_queue(cls, queue):
         """Construct a QuantumScript from an AnnotatedQueue."""
-        return QuantumScript(*process_queue(queue))
+        return cls(*process_queue(queue))
```

Comment:
What if the init for the sub class is different, would this break?

Reply:
yes it would! adding a test to block this rn

Reply:
see latest comment for details